src/pages/signup/index.tsx

```diff
@@ -1,5 +1,225 @@
-const Signup = () => (
-  <div>SignUp Page</div>
-);
+/* eslint-disable no-unused-vars */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { useState } from 'react';
+import {
+  Button, CssBaseline, TextField, Paper, Box, Grid, Typography, RadioGroup, FormControlLabel, Radio,
+} from '@mui/material';
+import { Link } from 'react-router-dom';
+import { useFormik } from 'formik';
+import validationSchema from './schema';
+import imageSrc from '../../assets/loginImg.jpg';
+import {
+  boxStyle, textFieldStyle, buttonStyle, gridStyle, imageStyle, fileUploadStyle,
+} from './classes';
+import './style.css';
+
+const handleFileUpload = (_event: React.ChangeEvent<HTMLInputElement>) => {
+  // eslint-disable-next-line no-console
+  console.log('File uploaded successfully.');
+};
+
+const Signup = () => {
+  const [userType, setUserType] = useState('');
+  const [showAdditionalFields, setShowAdditionalFields] = useState(false);
```

Comment:
No need for this as a state, actually we can render additional inputs based on user type

Reply:
done